src/game/Tactical/Dialogue_Control.cc

```diff
@@ -170,36 +170,52 @@ void ShutdownDialogueControl()
 	EmptyDialogueQueue();
 
 	// shutdown external static NPC faces
-	ShutdownStaticExternalNPCFaces();
+	UnloadExternalNPCFaces();
 
 	// gte rid of portraits for cars
 	UnLoadCarPortraits();
 }
 
 
-void InitalizeStaticExternalNPCFaces( void )
+void LoadExternalNPCFace(ProfileID mercID)
+{
+	if (externalNPCFaces.find(mercID) == externalNPCFaces.end())
+	{
+		externalNPCFaces[mercID] = &InitFace(mercID, nullptr, FACE_FORCE_SMALL);
+	}
+}
+
+FACETYPE* GetExternalNPCFace(ProfileID mercID)
+{
+	LoadExternalNPCFace(mercID); // ensure we have loaded the face
+	return externalNPCFaces.at(mercID);
+}
+
+void PreloadExternalNPCFaces()
 {
-	INT32 iCounter = 0;
 	// go and grab all external NPC faces that are needed for the game who won't exist as soldiertypes
 
 	if (fExternFacesLoaded) return;
 
 	fExternFacesLoaded = TRUE;
 
-	for( iCounter = 0; iCounter < NUMBER_OF_EXTERNAL_NPC_FACES; iCounter++ )
+	for (int i = 0; i < lengthof(preloadedExternalNPCFaces); i++)
```

Comment:
Why not just use .size()?

Reply:
`.size()` is a method in `std::array<ProfileID, 6>`. 

So we can either: 1) switch to `std::array<>` or 2) use `lengthof()`- which is sizeof(array) / sizeof(element).

Reply:
Ah, I see now I was looking at the definition of externalNPCFaces instead. Nevermind.